### PR TITLE
chore: remove checkIfProposal and checkKeys, adjust simpleExchange

### LINKED ISSUE #1522: Remove `checkKeys` and `checkIfProposal`
`checkKeys` is only still used by `checkIfProposal`. `checkIfProposal` is only still used by `simpleExchange`. `simpleExchange` uses `checkIfProposal` to support a pattern I doubt we want to encourage, and its use of `checkIfProposal` to support that pattern is questionable.

I'd prefer that both `checkKeys` and `checkIfProposal` be removed from zoeHelpers unless we think they still represent a reusable helper for a pattern that we still think is a good idea.

_Originally posted by @erights in https://github.com/Agoric/agoric-sdk/pull/1414#discussion_r471047549

### PR DESCRIPTION
**Pull request by katelynsills**
Closes #1522 

@erights does this match the direction you intended? Or was there another pattern you wanted?

### CODE REVIEW

File: packages/zoe/src/contracts/simpleExchange.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import '../../exported';
 import {
   swap,
   satisfies,
-  checkIfProposal,
+  assertProposalShape,
   assertIssuerKeywords,
 } from '../contractSupport/zoeHelpers';
```
Comment (erights) [inline packages/zoe/src/contracts/simpleExchange.js:12]: Line 66 below refers to "the handle for the matched offer".

File: packages/zoe/src/contracts/simpleExchange.js [hunk 2]
```diff
@@ -95,29 +95,46 @@ const start = zcf => {
...
   }
 
+  const sellAssetForPrice = harden({
+    give: { Asset: null },
+    want: { Price: null },
+  });
+
+  const sell = seat => {
+    buySeats = swapIfCanTradeAndUpdateBook(buySeats, sellSeats, seat);
+    bookOrdersChanged();
```
Comment (erights) [inline packages/zoe/src/contracts/simpleExchange.js]: You should move the call `bookOrdersChanged();` into `swapIfCanTradeAndUpdateBook`. That way there's no hazard of someone calling `swapIfCanTradeAndUpdateBook` and forgetting to notify.

File: packages/zoe/src/contracts/simpleExchange.js [hunk 3]
```diff
@@ -95,29 +95,46 @@ const start = zcf => {
...
-        `The proposal did not match either a buy or sell order.`,
-      );
+    // Buy Order
+    if (seat.getProposal().want.Asset) {
+      return buyHandler(seat);
     }
-    bookOrdersChanged();
-    return 'Trade Successful';
+    // Sell Order
+    if (seat.getProposal().give.Asset) {
```
Comment (erights) [inline packages/zoe/src/contracts/simpleExchange.js:129]: I wondered what you were going to test. Nice choice.

**Review by erights — APPROVED** [review]
LGTM

File: packages/zoe/src/contracts/simpleExchange.js (hunk 2 above)
Comment (katelynsills) [reply packages/zoe/src/contracts/simpleExchange.js]: Good idea

File: packages/zoe/src/contracts/simpleExchange.js (hunk 1 above)
Comment (katelynsills) [reply packages/zoe/src/contracts/simpleExchange.js:12]: ah, yep, I will update

**Pull request merged**